File: student-service/src/main/java/kz/baltabayev/studentservice/model/entity/Faculty.java
package kz.baltabayev.studentservice.model.entity;

import jakarta.persistence.*;
import kz.baltabayev.studentservice.model.enums.FacultyName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "faculties")
public class Faculty {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Enumerated(EnumType.STRING)
    private FacultyName name;
    private String dean;
}

File: faculty-department-service/src/main/kotlin/kz/baltabayev/facultydepartmentservice/model/entity/Faculty.kt
package kz.baltabayev.facultydepartmentservice.model.entity

import jakarta.persistence.*

@Entity
@Table(name = "faculty")
data class Faculty(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    var name: String? = null,
    var dean: String? = null,

    @OneToMany(fetch = FetchType.LAZY)
    var departments: List<Department> = listOf()
)